tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,974 @@
+    # how often to log sent message event
+    # On higher scale it could be >10000 to eliminate log IO overhead
+    consumer_logging_threshold: int = 1000
+    # How many workers will be in the message processing pool
+    worker_threads: int = 4
+    # web server port
+    web_port: int = 8090
+
+    @property
+    def topics_config(self):
+        """Generates topic config for TopicsConfig class
+
+        Returns:
+            dict: config that can be used for TopicsConfig(**topic_config)
+        """
+        return {
+            "topic_group_id": self.topic_group_id,
+            "topic_prefix": self.topic_prefix,
+            "topic_count": self.topic_count,
+            "consume_timeout": self.consume_timeout
+        }
+
+    @property
+    def forbidden_keys(self) -> list[str]:
+        # non-updatable vars via REST handle
+        return ['web_port', 'app_name']
+
+    def on_get(self, req, resp):
+        """Handles 'GET:<host:port>/' calls. Returns current app config
+
+        Args:
+            req (falcon.Request): request coming in
+            resp (falcon.Response): response to be filled
+        """
+        resp.status = falcon.HTTP_200
+        resp.content_type = falcon.MEDIA_JSON
+        # need to copy here otherwise, Class gets broken
+        _conf = deepcopy(vars(self))
+        # remove forbidden keys
+        for k in _conf.keys():
+            _conf.pop(k)
+        # dump it without prettify
+        resp.media = json.dumps(_conf)
+
+    def on_post(self, req, resp):
+        """Handles 'POST:<host:port>/' calls. Updates app config
+
+        Args:
+            req (falcon.Request): request coming in
+            resp (falcon.Response): response to be filled
+        """
+        # Get main logger from manager class
+        # If it exists, it will not be recreated
+        logger = setup_logger(LOGGER_MAIN)
+        logger.info(f"POST: Updating app configuration: {req.media}")
+        # App conf keys
+        keys_list = list(vars(self).keys())
+        # incoming keys
+        keys_req = list(req.media.keys())
+        # Validation
+        error_msgs = validate_keys(keys_req, keys_list, self.forbidden_keys)
+        if len(error_msgs) > 0:
+            # Just dump errors as list
+            msg = f"Incoming request invalid: {', '.join(error_msgs)}"
+            logger.error(msg)
+            resp.status = falcon.HTTP_400
+            resp.media_type = falcon.MEDIA_JSON
+            resp.media = {"errors": msg}
+        else:
+            # since validation is passed, no more stric checking needed
+            for k in req.media.keys():
+                setattr(self, k, req.media[k])
+            logger.debug(f"...updated app config: {vars(self)}")
+            resp.status = falcon.HTTP_200
+            resp.content_type = falcon.MEDIA_JSON
+            resp.media = {'result': "OK"}
+
+
+# singleton for app_config
+app_config = AppCfg()
+
+
+@dataclass(kw_only=True)
+class TopicsConfig:
```

Comment:
I'd rename it to WorkloadConfig or some such.. we are doing more than topic config here, example: consume_timeout is not a TopicConfig (in RP parlance)

Reply:
Make sense, yes